pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -340,19 +392,14 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *version.ServerV
 		return false, errors.Wrap(err, "set version")
 	}
 
-	if len(cr.Name) > clusterNameMaxLen {
-		return false, ErrClusterNameOverflow
+	err = cr.Validate()
+	if err != nil {
+		return false, errors.Wrap(err, "validate cr")
 	}
 
 	c := cr.Spec
-	if c.PXC == nil {
-		return false, fmt.Errorf("spec.pxc section is not specified. Please check %s cluster settings", cr.Name)
-	}
 
 	if c.PXC != nil {
-		if c.PXC.VolumeSpec == nil {
-			return false, fmt.Errorf("PXC: volumeSpec should be specified")
-		}
 		changed, err = c.PXC.VolumeSpec.reconcileOpts()
```

Comment:
can we also move `volume.resources.storage can't be empty` check from `reconcileOpts` into `Validate`?
for PXC, ProxySQL, Backup